ag_data/simulator.py

```diff
@@ -1,157 +1,36 @@
-        )
-
-    def createOrResetASensorTypeFromPresets(self, index):
-        """Create a sensor type object from available presets of sensor types
-
-        The sensor type record is a prerequisite for any sensor whose type is set to this.
-        The sensor type ID is also hardcoded in the database. Therefore, for the same sensor
-        type, if this method is called when the record exists, it will update the record.
-
-        Arguments:
-
-            index {int} -- the index of the sensor type preset to use.
-
-        Raises:
-
-            Exception: an exception raises when the index is not valid in presets.
-        """
-
-        if index > len(presets.sensor_type_presets) - 1:
-            raise Exception(
-                "Cannot find requested sensor type (index "
-                + str(index)
-                + ") from presets"
-            )
-        else:
-            pass
-
-        preset = presets.sensor_type_presets[index]
-
-        # If the sensor type record does not exist in the table, create the record.
-        record = models.AGSensorType.objects.filter(id=preset["agSensorTypeID"])
-
-        if record.count() == 0:
-            self.sensorType = models.AGSensorType.objects.create(
-                id=preset["agSensorTypeID"],
-                name=preset["agSensorTypeName"],
-                processing_formula=preset["agSensorTypeFormula"],
-                format=preset["agSensorTypeFormat"],
-            )
-        else:
-            record = record.first()
-            record.name = preset["agSensorTypeName"]
-            record.processing_formula = preset["agSensorTypeFormula"]
-            record.format = preset["agSensorTypeFormat"]
-            record.save()
-            self.sensorType = record
-
-    def createASensorFromPresets(self, index, cascadeCreation=False):
-        """Create a sensor from available presets of sensors
-
-        Arguments:
-
-            index {int} -- the index of the sensor preset to use.
-
-            cascadeCreation {bool=False} -- whether or not to create a corresponding sensor
-            type which the chosen sensor preset needs (default: {False})
-
-        Raises:
-
-            Exception: an exception raises when the index is not valid in presets.
-        """
+    def logProcessedSingleMeasurement(self, timestamp, value):
+        utilities.assertEvent(self.event)
+        utilities.assertSensor(self.sensor)
 
-        if index > len(presets.sensor_presets) - 1:
-            raise Exception(
-                "Cannot find requested sensor (index " + str(index) + ") from presets"
-            )
-        else:
-            pass
+        measurementDict = {
+            "measurement_timestamp": timestamp,
+            "measurement_sensor": self.sensor.id,
+            "measurement_values": value,
+        }
 
-        preset = presets.sensor_presets[index]
+        from ag_data.formulas.ingestion_engine import MeasurementIngestionEngine
```

Comment:
I think moving this to the top would be a better practice.
> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.


[https://www.python.org/dev/peps/pep-0008/#imports]()

Reply:
Will do.

Reply:
This is now in #376.